### PR TITLE
AP_Scripting: add some docs

### PR DESCRIPTION
**Pull request by magicrub**
Added scripting hooks `FWVersion:hal_board()` and `FWVersion:hal_board_subtype()`. I've also added suitable documentation to make the function calls more useful.

My inspiration was to check for
```
local HAL_BOARD_SITL = 3
if FWVersion:hal_board() == HAL_BOARD_SITL then
  -- do SITL specific handling
end
```

### CODE REVIEW

File: libraries/AP_Scripting/docs/docs.lua [hunk 1]
```diff
@@ -1145,14 +1145,44 @@ function FWVersion:minor() end
...
+-- here are some common ones:
+-- #define APM_BUILD_Rover          1
+-- #define APM_BUILD_ArduCopter     2
+-- #define APM_BUILD_ArduPlane      3
+-- #define APM_BUILD_AntennaTracker 4
+-- #define APM_BUILD_ArduSub        7
+-- #define APM_BUILD_AP_Periph      9
+-- #define APM_BUILD_Blimp          12
+-- #define APM_BUILD_Heli           13
 ---@return integer
```
Comment (IamPete1) [inline libraries/AP_Scripting/docs/docs.lua:1158]: There is markup for return types:
```suggestion
---@return integer
---| '1' Rover
---| '2' ArduCopter
---| '3' ArduPlane
---| '4' AntennaTracker
---| '7' ArduSub
---| '9' AP_Periph
---| '12' Blimp
---| '13' Heli
```

**Comment by IamPete1** [comment]
I wonder if a binding for `hal.util->get_system_id` would be more useful so you can get the full board string. I'm not sure how useful board and board subtype would be on real vehicles.

If the goal is just to detect SITL you can pick a random `SIM_` param to look for.

**Comment by magicrub** [comment]
I agree that the subtype is almost worthless...

**Comment by magicrub** [comment]
I've changed this PR to a NFC of just adding docs to existing ::type()

**Review by IamPete1 — APPROVED** [review]

**Pull request merged**